lib/kubeclient/watch_stream.rb

```diff
@@ -1,54 +1,63 @@
 require 'json'
-require 'net/http'
+require 'http'
 module Kubeclient
   module Common
     # HTTP Stream used to watch changes on entities
     class WatchStream
       def initialize(uri, http_options, format: :json)
         @uri = uri
-        @http = nil
-        @http_options = http_options.merge(read_timeout: nil)
+        @http_client = nil
+        @http_options = http_options
```

Comment:
@simon3z, `HTTP` does not time out by default so no explicit setup is needed:
 - https://github.com/httprb/http#timeouts
 - https://github.com/httprb/http/blob/v0.9.8/lib/http/options.rb#L10